dynamodb-stream-eventbridge-fanout/src/test/java/no/unit/nva/doi/publisher/EventBridgePublisherTest.java

```diff
@@ -0,0 +1,151 @@
+package no.unit.nva.doi.publisher;
```

Comment:
There is still noise in the class. I refactored it a bit, take a loot and see if the tests do what you want them to do. There are two commented out lines, just for you to be able to undo the change, because I am not sure if you need the successful entry in the failing attempt

```
  public static final String EXPECTED_DETAIL_TEMPLATE = "{\"eventSourceARN\":\"%s\"}";
    public static final String FAILED_EVENT_NAME = "Failed";
    public static final String SUCCESS_EVENT_NAME = "Success";
    public static final String RECORD_STRING_TEMPLATE = "{\"eventName\":\"%s\",\"eventSourceARN\":\"%s\"}";
    private static final String EVENT_BUS = UUID.randomUUID().toString();
    private static final Instant NOW = Instant.now();
    private static final String EVENT_SOURCE_ARN = UUID.randomUUID().toString();
    public static final Builder PUT_EVENT_REQUEST_BUILDER = putEventRequestBuilder();

    @Mock
    private EventBridgeRetryClient eventBridge;
    @Mock
    private EventPublisher failedEventPublisher;
    private EventPublisher publisher;

    /**
     * Set up environment for test.
     */
    @BeforeEach
    public void setup() {
        MockitoAnnotations.initMocks(this);

        publisher = new EventBridgePublisher(eventBridge, failedEventPublisher,
            EVENT_BUS, Clock.fixed(NOW, ZoneId.systemDefault()));
    }

    @Test
    public void publishCanPutEventsToEventBridge() {
        DynamodbEvent event = createDynamodbEvent();
        prepareMocksWithSuccessfulPutEvents();

        publisher.publish(event);

        PutEventsRequest expected = createPutEventsRequest();
        verify(eventBridge).putEvents(expected);
        verifyNoMoreInteractions(failedEventPublisher);
    }

    @Test
    public void publishFailedEventWhenPutEventsToEventBridgeHasFailures() {

        prepareMocksWithFailingPutEventEntries();

        DynamodbEvent.DynamodbStreamRecord failedRecord = createDynamodbStreamRecord(FAILED_EVENT_NAME);
        DynamodbEvent event = createDynamodbEvent( failedRecord);

        publisher.publish(event);

        PutEventsRequest partiallyFailingRequest = createFailingPutEventsRequest();
        verify(eventBridge).putEvents(partiallyFailingRequest);
        DynamodbEvent failedEvent = createDynamodbEvent(failedRecord);
        verify(failedEventPublisher).publish(failedEvent);
    }

    private static Builder putEventRequestBuilder() {
        return PutEventsRequestEntry.builder()
            .eventBusName(EVENT_BUS)
            .time(NOW)
            .source(EventBridgePublisher.EVENT_SOURCE)
            .detailType(EventBridgePublisher.EVENT_DETAIL_TYPE)
            .resources(EVENT_SOURCE_ARN);
    }

    private PutEventsRequest createFailingPutEventsRequest() {
       // String successRecordString = String.format(RECORD_STRING_TEMPLATE, SUCCESS_EVENT_NAME, EVENT_SOURCE_ARN);
        String failedRecordString = String.format(RECORD_STRING_TEMPLATE, FAILED_EVENT_NAME, EVENT_SOURCE_ARN);
        return PutEventsRequest.builder()
            .entries(//PUT_EVENT_REQUEST_BUILDER.detail(successRecordString).build(),
                PUT_EVENT_REQUEST_BUILDER.detail(failedRecordString).build())
            .build();
    }

    private List<PutEventsRequestEntry> createFailedEntries() {
        String failedRecordString = String.format(RECORD_STRING_TEMPLATE, FAILED_EVENT_NAME,
            EventBridgePublisherTest.EVENT_SOURCE_ARN);
        return Collections.singletonList(
            PUT_EVENT_REQUEST_BUILDER
                .detail(failedRecordString)
                .build());
    }

    private DynamodbEvent.DynamodbStreamRecord createDynamodbStreamRecord(String eventName) {
        DynamodbEvent.DynamodbStreamRecord record = new DynamodbEvent.DynamodbStreamRecord();
        record.setEventSourceARN(EVENT_SOURCE_ARN);
        record.setEventName(eventName);
        return record;
    }

    private PutEventsRequest createPutEventsRequest() {
        String expectedDetail = String.format(EXPECTED_DETAIL_TEMPLATE, EVENT_SOURCE_ARN);
        return PutEventsRequest.builder()
            .entries(PutEventsRequestEntry.builder()
                .eventBusName(EVENT_BUS)
                .time(NOW)
                .source(EventBridgePublisher.EVENT_SOURCE)
                .detailType(EventBridgePublisher.EVENT_DETAIL_TYPE)
                .detail(expectedDetail)
                .resources(EventBridgePublisherTest.EVENT_SOURCE_ARN)
                .build())
            .build();
    }

    private DynamodbEvent createDynamodbEvent() {
        DynamodbEvent.DynamodbStreamRecord record = new DynamodbEvent.DynamodbStreamRecord();
        record.setEventSourceARN(EVENT_SOURCE_ARN);
        return createDynamodbEvent(record);
    }

    private DynamodbEvent createDynamodbEvent(DynamodbStreamRecord... records) {
        DynamodbEvent event = new DynamodbEvent();
        event.setRecords(Arrays.asList(records));
        return event;
    }

    private void prepareMocksWithSuccessfulPutEvents() {
        when(eventBridge.putEvents(any(PutEventsRequest.class))).thenReturn(Collections.emptyList());
    }

    private void prepareMocksWithFailingPutEventEntries() {
        List<PutEventsRequestEntry> failedEntries = createFailedEntries();
        when(eventBridge.putEvents(any(PutEventsRequest.class))).thenReturn(failedEntries);
    }
```